src/Analyzers/CSharp/Tests/UseIndexOrRangeOperator/UseRangeOperatorTests.cs

```diff
@@ -842,5 +842,33 @@ void Goo(string s, int[] bar)
                 FixedCode = fixedSource,
             }.RunAsync();
         }
+
+        [WorkItem(47183, "https://github.com/dotnet/roslyn/issues/47183")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseRangeOperator)]
+        public async Task TestExpressionWithNullConditionalAccess()
+        {
+            var source =
+@"
+#nullable enable
+public class Test
+{
+    public string? M(string? arg)
+        => arg?.Substring([|42|]);
+}";
+            var fixedSource =
+@"
+#nullable enable
+public class Test
+{
+    public string? M(string? arg)
+        => arg?[42..];
+}";
+            await new VerifyCS.Test
+            {
+                ReferenceAssemblies = ReferenceAssemblies.NetCore.NetCoreApp31,
+                TestCode = source,
+                FixedCode = fixedSource,
+            }.RunAsync();
+        }
```

Comment:
can we have a test for cases like:

`a?.b.Substring(42)`  and
`a?.Substring(42).Length`

?

Thanks!

Reply:
Done in cddd46ce6019fb5e82c95e8608eee821e845f75e

Reply:
i don't see the test for `a?.Substring(42).Length`, can you link me to the line for that?

Reply:
(note, conditional access, and property access in teh same expr).